### PR TITLE
example/elf : Add mem fault cases for mpu test

### PR DESCRIPTION
**Pull request by jeongarmy**
- examples/elf: Add memfault cases for mpu test
- imxrt1050-evk/loadable_elf_apps: set configuration of address for mpu test

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a87482893ae234ef5bc27f747c1ebba74e346b55] - Code Rule Check OK.

File: apps/examples/elf/wifiapp/wifiapp.c [hunk 1]
```diff
@@ -19,19 +19,44 @@
...
 #include <unistd.h>
 #include <errno.h>
 #include <sched.h>
 #include <pthread.h>
 
+#if !defined(CONFIG_MPU_TEST_KERNEL_CODE_ADDR) || !defined(CONFIG_MPU_TEST_APP_ADDR)
+#error "Address not defined for MPU test"
+#endif
+
 static void *assert_thread(void *index)
```
Comment (sijupunnnoose) [inline apps/examples/elf/wifiapp/wifiapp.c]: Not using index anymore, remove it?

**Comment by seinfra** [comment]
Target : [b2d747f1f17ef8140d4fc5910da8f2087ef2f8e8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b2d747f1f17ef8140d4fc5910da8f2087ef2f8e8] - Code Rule Check OK.

File: apps/examples/elf/wifiapp/wifiapp.c [hunk 2]
```diff
@@ -19,19 +19,44 @@
...
 {
-	printf("[%d] %dth thread, assert!\n", getpid(), (int)index);
+	int type;
+	uint32_t dest;
+	volatile uint32_t * addr;
```
Comment (sunghan-chang) [inline apps/examples/elf/wifiapp/wifiapp.c]: Let's remove a space

**Comment by seinfra** [comment]
Target : [86da8454932463465225fbd4c0c78ce70467964a] -

**Comment by seinfra** [comment]
Target : [86da8454932463465225fbd4c0c78ce70467964a] - Code Rule Check (C++) OK.

File: apps/examples/elf/wifiapp/wifiapp.c [hunk 3]
```diff
@@ -19,19 +19,44 @@
...
+		printf("[%d] %dth thread, PANIC!\n", getpid(), (int)index);
+		sleep(1);
+		PANIC();
+	} else if (type == 1) {
+		/* Access kernel code */
+		addr = CONFIG_MPU_TEST_KERNEL_CODE_ADDR;
+		printf("[%d] %dth thread, Write kernel code space 0x%x\n", getpid(), (int)index, addr);
+		sleep(1);
+		*addr = 0xdeadbeef;
+	} else if (type == 2) {
```
Comment (sunghan-chang) [inline apps/examples/elf/wifiapp/wifiapp.c]: else

**Comment by seinfra** [comment]
Target : [c644580787e13a1c12e438fbaa503a8203a5bc8d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c644580787e13a1c12e438fbaa503a8203a5bc8d] - Code Rule Check (C++) OK.

**Pull request merged**